specification/security/resource-manager/common/v1/types.json

```diff
@@ -242,6 +247,30 @@
         "machineName"
       ]
     },
+    "OnPremiseSqlResourceDetails": {
+      "type": "object",
+      "description": "Details of the On Premise Sql resource that was assessed",
+      "x-ms-discriminator-value": "OnPremiseSql",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OnPremiseResourceDetails"
```

Comment:
Make sure this inherits correctly, and doesn't just defaults to base type

Reply:
I implemented it this way after seeing similar examples, for example :
Take a look at resource AllowlistCustomAlertRule that inherited from ListCustomAlertRule that inherited from CustomAlertRule at file:

https://github.com/Azure/azure-rest-api-specs/blob/51f9f419551370294dec446c033e942208e6b023/specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/deviceSecurityGroups.json


The SDK that generated from the swagger -
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/securitycenter/Microsoft.Azure.Management.SecurityCenter/src/Generated/Models/ListCustomAlertRule.cs
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/securitycenter/Microsoft.Azure.Management.SecurityCenter/src/Generated/Models/AllowlistCustomAlertRule.cs